docs/react-v9/contributing/testing-with-jest.md

```diff
@@ -0,0 +1,234 @@
+> 🚨🚨 **This page is primarily about `@fluentui/react-components` ("v9") and `@fluentui/react` ("v8") and related packages.** 🚨🚨&nbsp; [See this page for `@fluentui/react-northstar` ("v0").](https://github.com/microsoft/fluentui/blob/master/packages/fluentui/test-a-feature.md)
+
+## Overview
+
+Fluent UI's unit, functional, and snapshot tests are built using [Jest](https://jestjs.io/). This allows us to run tests in a Node environment but simulates the browser using [jsdom](https://github.com/jsdom/jsdom). ([See this page for other types of tests.](Testing))
+
+We use various libraries on top of Jest for rendering and interacting with React components:
+
+- New tests should use [React Testing Library](https://testing-library.com/docs/react-testing-library/intro)
+- Legacy tests may also use:
+  - [Enzyme](https://enzymejs.github.io/enzyme/) -- **not allowed in new tests**
+  - [react-test-renderer](https://reactjs.org/docs/test-renderer.html), primarily for snapshot testing _(snapshots are discouraged in new tests)_
+
+## Running tests
+
+Our Jest setup generally require that packages be built before testing, so before running tests the first time, you should run `yarn build --to my-package` from the repo root.
+
+To run the tests for one package, `cd` to that package and then run `yarn test`.
+
+To run an individual test (or technically any tests with a path matching this substring), `cd` to the relevant package and run `yarn jest MyTestName`.
```

Comment:
pls remove, and point folks to jest documentation. In general I'd suggest that contributors use `watch` mode where they can narrow down test run from CLI - betrer DX

```suggestion
```